apps/src/javalab/JavalabEditorTabMenu.jsx

```diff
@@ -27,19 +31,43 @@ export class JavalabEditorTabMenu extends Component {
     const {
       renameFromTabMenu,
       deleteFromTabMenu,
+      moveTabRight,
+      moveTabLeft,
       showVisibilityOption,
       changeFileTypeFromTabMenu,
       fileIsVisible,
-      fileIsValidation
+      fileIsValidation,
+      activeTabKey,
+      orderedTabKeys
     } = this.props;
+
     let elements = [
       <button onClick={renameFromTabMenu} key="rename" type="button">
         {javalabMsg.rename()}
-      </button>,
+      </button>
+    ];
+    const tabsLength = orderedTabKeys.length;
+    const index = orderedTabKeys.indexOf(activeTabKey);
+    if (index > 0) {
+      elements.push(
+        <button onClick={moveTabLeft} key="moveLeft" type="button">
+          {javalabMsg.moveLeft()}
+        </button>
+      );
+    }
+    if (index < tabsLength - 1) {
+      elements.push(
+        <button onClick={moveTabRight} key="moveRight" type="button">
+          {javalabMsg.moveRight()}
+        </button>
+      );
+    }
+    elements.push(
       <button onClick={deleteFromTabMenu} key="delete" type="button">
         {javalabMsg.delete()}
       </button>
-    ];
+    );
+
```

Comment:
There's interesting enough logic here that adding some unit tests would be valuable, especially to make sure that moveLeft/right are hidden at the appropriate times.

Reply:
Sounds good - will do!